### PR TITLE
Show softfork status in getblockchaininfo

### PR DESCRIPTION
**Pull request by laanwj**
```
  "softforks": [            (array) status of softforks in progress
     {
        "id": "xxxx",        (string) name of softfork
        "version": xx,         (numeric) block version
        "enforce": {           (object) progress toward enforcing the softfork rules
           "status": xx,       (boolean) true if threshold reached
           "found": xx,        (numeric) number of blocks with the new version found
           "required": xx,     (numeric) number of blocks required to trigger
           "window": xx,       (numeric) size of examined window of recent blocks
        },
        "reject": { ... }      (object) progress toward rejecting pre-softfork blocks (same fields as for "enforce")
     }, ...
  ]
```

### CODE REVIEW

**Comment by btcdrak** [comment]
Tested ACK

**Comment by petertodd** [comment]
Tested ACK.

One minor nit is the way the "found" bit caps out at found==required. I *think* removing the "&& nFound < nRequired" clause in the for loop should be safe; the nFound > nRequired part at the end should never be triggered. But just documenting it would be safer.

**Comment by laanwj** [comment]
> One minor nit is the way the "found" bit caps out at found==required. I think removing the "&& nFound < nRequired" clause in the for loop should be safe; the nFound > nRequired part at the end should never be triggered. But just documenting it would be safer.

I've thought about cloning the `IsSuperMajority` in rpcblockhain.cpp, with that change, leaving the original one as-is. It is only a few lines. Then it also doesn't need to be exported from main anymore.

**Comment by btcdrak** [comment]
@laanwj Sounds like a good solution.

**Comment by petertodd** [comment]
@laanwj ACK that.

**Comment by jtimon** [comment]
ut ACK

**Comment by CodeShark** [comment]
tACK

File: src/rpcblockchain.cpp [hunk 1]
```diff
@@ -469,6 +469,36 @@ UniValue verifychain(const UniValue& params, bool fHelp)
     return CVerifyDB().VerifyDB(pcoinsTip, nCheckLevel, nCheckDepth);
 }
 
+/* Implementation of IsSuperMajority with better feedback */
```
Comment (fanquake) [inline src/rpcblockchain.cpp]: nit: /**

**Comment by fanquake** [comment]
utACK

File: src/rpcblockchain.cpp (hunk 1 above)
Comment (laanwj) [inline src/rpcblockchain.cpp]: Fixed

File: src/rpcblockchain.cpp [hunk 2]
```diff
@@ -501,6 +544,14 @@ UniValue getblockchaininfo(const UniValue& params, bool fHelp)
     obj.push_back(Pair("verificationprogress",  Checkpoints::GuessVerificationProgress(Params().Checkpoints(), chainActive.Tip())));
     obj.push_back(Pair("chainwork",             chainActive.Tip()->nChainWork.GetHex()));
     obj.push_back(Pair("pruned",                fPruneMode));
+
+    const Consensus::Params& consensusParams = Params().GetConsensus();
```
Comment (jonasschnelli) [inline src/rpcblockchain.cpp]: Now we have the soft fork states between `pruned` and `prunehight` which looks a bit sandwiched.

**Pull request merged**